src/engine/CountAvailablePredicates.cpp

```diff
@@ -91,12 +91,12 @@ size_t CountAvailablePredicates::getSizeEstimate() {
     // for the type of optimizations the optimizer can currently do.
     size_t num_distinct = _subtree->getSizeEstimate() /
                           _subtree->getMultiplicity(_subjectColumnIndex);
-    return num_distinct / getIndex().getHasPredicateMultiplicityPredicates();
+    return num_distinct / getIndex().getAvgNumDistinctSubjectsPerPredicate();
```

Comment:
The estimate for the `else` case is now perfect. But shouldn't the estimate for the general case also be very good when the part constraining the subject is just a single triple, where almost all subjects match and we know the exact statistics from the index scan? However, the size estimate is way off (much too low) in that case. Here is an example query: https://qlever.cs.uni-freiburg.de/osm-germany/3m6qtD